accession-commons-monotonic-generator-jpa/src/test/java/uk/ac/ebi/ampt2d/commons/accession/core/BasicMonotonicTestForAlternateRanges.java

```diff
@@ -0,0 +1,111 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.commons.accession.core;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+import uk.ac.ebi.ampt2d.commons.accession.core.exceptions.AccessionCouldNotBeGeneratedException;
+import uk.ac.ebi.ampt2d.commons.accession.generators.DecoratedAccessionGenerator;
+import uk.ac.ebi.ampt2d.commons.accession.generators.monotonic.MonotonicAccessionGenerator;
+import uk.ac.ebi.ampt2d.commons.accession.hashing.SHA1HashingFunction;
+import uk.ac.ebi.ampt2d.commons.accession.persistence.jpa.monotonic.service.ContiguousIdBlockService;
+import uk.ac.ebi.ampt2d.commons.accession.service.BasicMonotonicAccessioningService;
+import uk.ac.ebi.ampt2d.test.TestModel;
+import uk.ac.ebi.ampt2d.test.configuration.TestMonotonicDatabaseServiceTestConfiguration;
+import uk.ac.ebi.ampt2d.test.service.TestMonotonicDatabaseService;
+
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@ContextConfiguration(classes = {TestMonotonicDatabaseServiceTestConfiguration.class})
+@TestPropertySource("classpath:application-test1.properties")
+public class BasicMonotonicTestForAlternateRanges {
+
+    @Autowired
+    private TestMonotonicDatabaseService databaseService;
+    @Autowired
+    private ContiguousIdBlockService contiguousIdBlockService;
+
+    private static final int BLOCK_SIZE = 5;
+    private static final String INSTANCE_ID = "APP1";
+    private static final long NEXT_BLOCK_INTERVAL = 5L;
+    private static final String CATEGORY_ID = "eva";
+
+    @Test
+    public void testAlternateRanges() throws AccessionCouldNotBeGeneratedException {
+        BasicAccessioningService<TestModel, String, Long> accessioningService = getAccessioningService1();
+
+        List<AccessionWrapper<TestModel, String, Long>> evaAccessions = accessioningService.getOrCreate(
+                Arrays.asList(
+                        TestModel.of("service-test-1"),
+                        TestModel.of("service-test-2"),
+                        TestModel.of("service-test-3"),
+                        TestModel.of("service-test-4"),
+                        TestModel.of("service-test-5"),
+                        TestModel.of("service-test-6")
+                ));
+
+        assertEquals(6, evaAccessions.size());
+        evaAccessions.stream().allMatch(accession -> accession.getAccession() < 5L || accession.getAccession() == 11L);
```

Comment:
This will only return true or false, not assert anything. And I don't think it will be much clear if you do an assertTrue over the result of that.